oeps/oep-0001.rst

```diff
@@ -0,0 +1,318 @@
+OEPs are not the only way for a change to be made to Open edX,
+however the goal is for it to be a repository collecting the
+history of large and/or broadly relevant choices. 
+
+OEP Types
+=========
+* A Process proposal describes a change to how the Open edX community
+  functions.
+* A "Best Practices" proposal describes a technology or implementation
+  choice that the Open edX community believes all Open edX services
+  and/or libraries should use or follow.
+* An Architecture proposal describes relationships between Open edX
+  services and libraries, including their APIs and data formats.
+
+Workflow
+========
+
+Submitting an OEP
+-----------------
+The OEP process begins with a new idea for Open edX. It is highly
+recommended that a single OEP contain a single key proposal or new idea.
+Small enhancements or patches often don't need an OEP and can be injected
+into the Open edX development workflow with a patch submission.
+
+Each OEP must have a champion -- someone who writes the OEP using the
+style and format described below, shepherds the discussions in the
+appropriate forums, and attempts to build community consensus around
+the idea. The OEP champion (a.k.a. Author) should first attempt to
+ascertain whether the idea is appropriate for an OEP. Posting to the
+`edx-code`_ mailing list, ``arch@edx.org`` and ``archexternal@edx.org``
+is the best way to go about this.
+
+Vetting an idea publicly before going as far as writing an OEP is meant
+to save the potential author time. Many ideas have been brought forward
+for changing Open edX that have been rejected for various reasons. Asking
+the Open edX community first if an idea is original helps prevent too much
+time being spent on something that is guaranteed to be rejected based on
+prior discussions (searching the internet does not always do the trick).
+It also helps to make sure the idea is applicable to the entire community
+and not just the author. Just because an idea sounds good to the author
+does not mean it will work for most Open edX installations.
+
+Once the champion has asked the Open edX community as to whether an idea
+has any chance of acceptance, a draft OEP should be submitted as a pull request
+against the `central OEP repository`_. The pull request title must start
+with "OEP-XXX", and claims that OEP number for the included proposal.
+Before claiming a number, check the list of previous pull requests to
+see which numbers have already been assigned.
+
+.. _central OEP repository: http://github.com/edx/open-edx-proposals
+
+As updates are necessary, the OEP author can update the pull request.
+
+OEPs can be reference pull requests to other Open edX repositiories which
+will act as reference implementations.
+
+OEP Review & Resolution
+-----------------------
+
+Once the authors have completed an OEP, they may request a review from
+OEP reviewers. These reviewers will select one member to act as the final
```

Comment:
Oops. That terminology is a holdover from an earlier version. For now, in
sticking with just having Eddie decide, and not having an explicit group of
reviewers (the entire community is both the pool of reviewers and the pool
of potential Arbiters).

On Wed, May 11, 2016, 9:38 AM Gabe Mulley <notifications@github.com> wrote:

> In oeps/oep-0001.rst
> <https://github.com/cpennington/open-edx-proposals/pull/1#discussion_r62848671>
> :
>
> > +with "OEP-XXX", and claims that OEP number for the included proposal.
> > +Before claiming a number, check the list of previous pull requests to
> > +see which numbers have already been assigned.
> > +
> > +.. _central OEP repository: http://github.com/edx/open-edx-proposals
> > +
> > +As updates are necessary, the OEP author can update the pull request.
> > +
> > +OEPs can be reference pull requests to other Open edX repositiories which
> > +will act as reference implementations.
> > +
> > +OEP Review & Resolution
> > +-----------------------
> > +
> > +Once the authors have completed an OEP, they may request a review from
> > +OEP reviewers. These reviewers will select one member to act as the final
>
> In "OEP Roles" we say "The Arbiter will be chosen by Eddie Fagin", but
> here we say they are chosen by the reviewers. I personally prefer the idea
> of having the reviewers choose an arbiter from their midst, but I thought I
> would at least point out the inconsistency.
>
> Also - it's unclear to me how the reviewers are chosen.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cpennington/open-edx-proposals/pull/1/files/0dc13cdd4c64f24f01e5cf9ccafe475884ddc3b7#r62848671>
>